src/GridFS/WritableStream.php

```diff
@@ -283,7 +283,7 @@ private function insertChunkFromBuffer(): void
         $chunk = [
             'files_id' => $this->file['_id'],
             'n' => $this->chunkOffset,
-            'data' => new Binary($data, Binary::TYPE_GENERIC),
+            'data' => new Binary($data),
```

Comment:
The psalm error may be due to outdated stubs. 

https://github.com/JetBrains/phpstorm-stubs/blob/master/mongodb/BSON/Binary.php#L34

Strangely, it only alerts on this line, whereas it should be the same everywhere you've modified the call to `new Binary`.

I propose that you update the baseline until PHPORM-25.

Reply:
Aye, forgot to run CS tools locally. I'll get this fixed up.

Reply:
Just noticed that the [Binary stub](https://github.com/JetBrains/phpstorm-stubs/blob/master/mongodb/BSON/Binary.php) was never updated for [PHPC-2101](https://jira.mongodb.org/browse/PHPC-2101). I opened [PHPORM-40](https://jira.mongodb.org/browse/PHPORM-40) to track that, as there might have been other things missed in v1.15.

https://jira.mongodb.org/browse/PHPORM-40

Reply:
@GromNaN: What does PhpStorm have to do with Psalm?

The [Psalm stub for `ext-mongodb`](https://github.com/vimeo/psalm/blob/master/stubs/extensions/mongodb.phpstub) only defines a few things with generics. I don't see anything in [Psalm's `composer.json`](https://github.com/vimeo/psalm/blob/master/composer.json) that pulls in PhpStorm stubs.

I was under the impression that Psalm inferred extension APIs via Reflection, which should be [correct](https://github.com/mongodb/mongo-php-driver/blob/1.16.1/src/BSON/Binary.stub.php#L66).

Reply:
Exact. Sorry for the confusion.